### PR TITLE
[FRR Patch] Added patch in FRR to send tag value associated with route via Netlink to fpmsyncd

### PR DESCRIPTION
**Pull request by abdosi**
What I did: 
Added patch in FRR to send tag value associated with route via NETLINK RTA_PRIORITY field which can be used as attribute/metadata in fpmsyncd for different use-cases.

Why I did:
Some of use cases:
1. Do not program specific routes into APP_DB
2. Fallback to default route for specific routes.

How I verify:
Manual Verification.

### CODE REVIEW

**Review by prsunny — COMMENTED** [review]
lgtm

**Comment by prsunny** [comment]
@StormLiangMS for viz

File: src/sonic-frr/patch/0054-Added-Change-in-Zebra-FPM-library-to-send-tag-value-.patch [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+index 231047143..2b45087d4 100644
+--- a/zebra/zebra_fpm_netlink.c
++++ b/zebra/zebra_fpm_netlink.c
+@@ -297,7 +297,8 @@ static int netlink_route_info_fill(struct netlink_route_info *ri, int cmd,
+ 
+ 	ri->rtm_protocol = netlink_proto_from_route_type(re->type);
+ 	ri->rtm_type = RTN_UNICAST;
+-	ri->metric = &re->metric;
++	// Patch to send tag value via NETLINK Priority field (RTA_PRIORITY). The Tag vale can be used as metadata/attribute in fpmsyncd for further processing
++	ri->metric = &re->tag;
```
Comment (dgsudharsan) [inline src/sonic-frr/patch/0054-Added-Change-in-Zebra-FPM-library-to-send-tag-value-.patch:23]: Given now we migrated to dplane_fpm_sonic, does it require any changes over there?

File: src/sonic-frr/patch/0054-Added-Change-in-Zebra-FPM-library-to-send-tag-value-.patch [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+diff --git a/zebra/zebra_fpm_netlink.c b/zebra/zebra_fpm_netlink.c
+index 231047143..2b45087d4 100644
+--- a/zebra/zebra_fpm_netlink.c
++++ b/zebra/zebra_fpm_netlink.c
+@@ -297,7 +297,8 @@ static int netlink_route_info_fill(struct netlink_route_info *ri, int cmd,
+ 
+ 	ri->rtm_protocol = netlink_proto_from_route_type(re->type);
+ 	ri->rtm_type = RTN_UNICAST;
+-	ri->metric = &re->metric;
```
Comment (dgsudharsan) [inline src/sonic-frr/patch/0054-Added-Change-in-Zebra-FPM-library-to-send-tag-value-.patch:21]: We are overriding the current meaning of metric by default. Would it impact any existing usecases/ scenarios?

File: src/sonic-frr/patch/0054-Added-Change-in-Zebra-FPM-library-to-send-tag-value-.patch (hunk 1 above)
Comment (abdosi) [reply src/sonic-frr/patch/0054-Added-Change-in-Zebra-FPM-library-to-send-tag-value-.patch]: @dgsudharsan thanks.  i have revised the patch. Change is now inside API `netlink_route_multipath_msg_encode` which is called by both `dplane_fpm_sonic.c` and `dplane_fpm_nl.c`

File: src/sonic-frr/patch/0054-Added-Change-in-Zebra-FPM-library-to-send-tag-value-.patch (hunk 2 above)
Comment (abdosi) [reply src/sonic-frr/patch/0054-Added-Change-in-Zebra-FPM-library-to-send-tag-value-.patch]: @dgsudharsan not applicable as change has been revised. please review again.

**Comment by abdosi** [comment]
@dgsudharsan : can you review again

**Review by dgsudharsan — APPROVED** [review]
@abdosi  I request you to wait for FRR 10.0.1 merge and rebase this PR based on it. @hasan-brcm  FYI

**Comment by abdosi** [comment]
> @abdosi I request you to wait for FRR 10.0.1 merge and rebase this PR based on it. @hasan-brcm FYI

@dgsudharsan : Please approve again.

**Review by rlhui — APPROVED** [review]

**Pull request merged**